pkg/codegen/nodejs/gen.go

```diff
@@ -934,19 +934,19 @@ func (mod *modContext) genResource(w io.Writer, r *schema.Resource) (resourceFil
 		methodName := camel(method.Name)
 		fun := method.Function
 
-		var objectReturnType *schema.ObjectType
-		if fun.ReturnType != nil {
-			if objectType, ok := fun.ReturnType.(*schema.ObjectType); ok && objectType != nil {
-				objectReturnType = objectType
-			} else if !fun.ReturnTypePlain {
-				// Currently the code only knows how to generate code for methods returning an
-				// ObjectType or methods returning a plain resource All other methods are simply
-				// skipped; bail here.
-				return
-			}
-		}
+		// The return type may be lifted from a single value based on the language schema property
+		// "liftSingleValueMethodReturns", Or it may simply be a plain scalar type.  In each case we
+		// need to handle code generation slightly differently. When it is lifted we generate a
+		// function that accesses the single value property of the object, when it is plain the
+		// returning object has a single "result__" property.
```

Comment:
Is this true any more? I thought we basically just say "if lifted at schema level or function level, generate a `callSingle` call".